### PR TITLE
150 extend the copyright script for python files

### LINKED ISSUE #150: Extend the copyright script for python files
#### Description
Currently the script `scripts/update_copyright.py` only changes `.cpp` and `.h` files. It should also consider python and shell files

#### Questions/Ideas
- The comment operator depends on the extension:
  - `.py`, `.sh` : `#`
  - `.cmd`, `.bat` : `REM`
  - `.h`, `.cpp`: `//`

### PR DESCRIPTION
**Pull request by folmos-at-orange**

### CODE REVIEW

File: .cmake-format.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Copyright (c) 2024 Orange. All rights reserved.
```
Comment (popescu-v) [inline .cmake-format.py:1]: s/bannor/banner/ in commit message for https://github.com/KhiopsML/khiops/pull/363/commits/5cdbba9b0097166236412081701004e7685ed22c

File: scripts/update-copyright.py [hunk 2]
```diff
@@ -1,44 +1,74 @@
...
 def update_copyright(file_path):
-    """Updates the copyright notice of a file"""
-    print(f"Updating {file_path}")
+    """Updates the copyright notice of a file if necessary"""
+
+    # Obtain the comment prefix string from the file extension
+    _, ext = os.path.splitext(file_path)
+    if ext in (".h", ".c", ".hpp", ".cpp", ".java"):
+        comment_prefix = b"// "
+    elif ext == ".py":
```
Comment (popescu-v) [inline scripts/update-copyright.py:53]: Perhaps we should do the same for Bash scripts as well (`khiops-env`, `khiops`, `khiops_coclustering`) and their Windows counterparts?

File: test/MemoryStatsVisualizer/collect_stats_per_task.py [hunk 3]
```diff
@@ -1,3 +1,7 @@
+# Copyright (c) 2024 Orange. All rights reserved.
+# This software is distributed under the BSD 3-Clause-clear License, the text of which is available
+# at https://spdx.org/licenses/BSD-3-Clause-Clear.html or see the "LICENSE" file for more details.
+
 # -*- coding: utf-8 -*-
```
Comment (popescu-v) [inline test/MemoryStatsVisualizer/collect_stats_per_task.py]: AFAIK, this special comment should be the first line in the file, if we keep it. But for Python 3, this is not necessary anymore AFAIK. So: either remove it, or add special logic to keep at the top of the file.

The same applies to all files that are concerned.

**Review by popescu-v — CHANGES_REQUESTED** [review]
See comments.

File: scripts/update-copyright.py (hunk 2 above)
Comment (folmos-at-orange) [reply scripts/update-copyright.py:53]: The issue stated that we'll do it only for `.py`. It was discussed in a weekly.

File: test/MemoryStatsVisualizer/collect_stats_per_task.py (hunk 3 above)
Comment (folmos-at-orange) [reply test/MemoryStatsVisualizer/collect_stats_per_task.py]: Ok, it seems that the standard encoding is UTF8 so i'll remove these.

File: scripts/update-copyright.py (hunk 2 above)
Comment (popescu-v) [reply scripts/update-copyright.py:53]: Ok, I understand your point. However the issue (#150) states "Extend the copyright script for python and script files". So, I would edit the issue description according to what was convened (I can see the comment to the issue).

File: test/MemoryStatsVisualizer/collect_stats_per_task.py (hunk 3 above)
Comment (popescu-v) [reply test/MemoryStatsVisualizer/collect_stats_per_task.py]: Ok, that special comment was only for Python 2 (standard encoding was ASCII AFAIR); not needed for Python 3.

File: scripts/update-copyright.py (hunk 2 above)
Comment (folmos-at-orange) [reply scripts/update-copyright.py:53]: Done

File: test/MemoryStatsVisualizer/collect_stats_per_task.py (hunk 3 above)
Comment (folmos-at-orange) [reply test/MemoryStatsVisualizer/collect_stats_per_task.py]: Done

**Review by popescu-v — APPROVED** [review]

**Pull request merged**